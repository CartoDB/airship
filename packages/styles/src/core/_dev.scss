@import 'tools/_mixins';
@import 'variables/_variables';
