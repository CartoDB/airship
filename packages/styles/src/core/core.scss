@import 'variables/_variables';

* {
  box-sizing: border-box;
  font-family: sans-serif;
}

:root {
  --primary-color: #{$primary-color};
}
