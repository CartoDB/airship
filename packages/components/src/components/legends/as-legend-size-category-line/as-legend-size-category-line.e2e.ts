import { E2EPage, newE2EPage } from '@stencil/core/testing';

describe('as-legend-size-category-line', async () => {
  describe('Rendering', async () => {
    let page: E2EPage;

    it('basic', async () => {
      page = await newE2EPage({ html: `
        <as-legend-size-category-line>
        </as-legend-size-category-line>
      ` });

      const element = await page.find('as-legend-size-category-line');
      element.setProperty('data', [{
        label: 'point label',
        width: 10,
      },
      {
        label: 'line label',
        width: 12,
      },
      {
        label: 'polygon label',
        width: 14,
      }]);
      await page.waitForChanges();
      expect(element.outerHTML).toMatchSnapshot();
    });

    it('basic horizontal', async () => {
      page = await newE2EPage({ html: `
        <as-legend-size-category-line>
        </as-legend-size-category-line>
      ` });

      const element = await page.find('as-legend-size-category-line');
      element.setProperty('data', [{
        label: 'point label',
        width: 10,
      },
      {
        label: 'line label',
        width: 12,
      },
      {
        label: 'polygon label',
        width: 14,
      }]);
      element.setProperty('orientation', 'horizontal');
      await page.waitForChanges();
      expect(element.outerHTML).toMatchSnapshot();
    });

    it('renders line styles', async () => {
      page = await newE2EPage({ html: `
        <as-legend-size-category-line>
        </as-legend-size-category-line>
      ` });

      const element = await page.find('as-legend-size-category-line');
      element.setProperty('data', [{
        label: 'point label',
        strokeStyle: 'dashed',
        width: 10,
      },
      {
        label: 'line label',
        strokeStyle: 'dotted',
        width: 12,
      },
      {
        label: 'polygon label',
        strokeStyle: 'dashed',
        width: 14,
      }]);
      await page.waitForChanges();
      expect(element.outerHTML).toMatchSnapshot();
    });
  });
});
