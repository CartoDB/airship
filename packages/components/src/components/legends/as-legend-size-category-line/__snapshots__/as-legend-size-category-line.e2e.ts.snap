// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`as-legend-size-category-line Rendering basic 1`] = `"<as-legend-size-category-line class=\\"hydrated\\"> <div class=\\"as-legend-size-category-line--steps as-legend-size-category-line--vertical\\"><div class=\\"as-legend-size-category-line--step\\"><div class=\\"as-legend-size-category-line--line-wrapper\\" style=\\"width: 48px;\\"><div class=\\"as-legend-size-category-line--line\\" style=\\"height: 10px; width: 48px; border-top-style: solid; border-top-width: 10px;\\"></div></div><span class=\\"as-legend-size-category-line--label\\">point label</span></div><div class=\\"as-legend-size-category-line--step\\"><div class=\\"as-legend-size-category-line--line-wrapper\\" style=\\"width: 48px;\\"><div class=\\"as-legend-size-category-line--line\\" style=\\"height: 12px; width: 48px; border-top-style: solid; border-top-width: 12px;\\"></div></div><span class=\\"as-legend-size-category-line--label\\">line label</span></div><div class=\\"as-legend-size-category-line--step\\"><div class=\\"as-legend-size-category-line--line-wrapper\\" style=\\"width: 48px;\\"><div class=\\"as-legend-size-category-line--line\\" style=\\"height: 14px; width: 48px; border-top-style: solid; border-top-width: 14px;\\"></div></div><span class=\\"as-legend-size-category-line--label\\">polygon label</span></div></div></as-legend-size-category-line>"`;

exports[`as-legend-size-category-line Rendering basic horizontal 1`] = `"<as-legend-size-category-line class=\\"hydrated\\"> <div class=\\"as-legend-size-category-line--steps as-legend-size-category-line--horizontal\\"><div class=\\"as-legend-size-category-line--step\\"><div class=\\"as-legend-size-category-line--line-wrapper\\" style=\\"height: 48px;\\"><div class=\\"as-legend-size-category-line--line\\" style=\\"height: 10px; width: 48px; border-top-style: solid; border-top-width: 10px;\\"></div></div><span class=\\"as-legend-size-category-line--label\\">point label</span></div><div class=\\"as-legend-size-category-line--step\\"><div class=\\"as-legend-size-category-line--line-wrapper\\" style=\\"height: 48px;\\"><div class=\\"as-legend-size-category-line--line\\" style=\\"height: 12px; width: 48px; border-top-style: solid; border-top-width: 12px;\\"></div></div><span class=\\"as-legend-size-category-line--label\\">line label</span></div><div class=\\"as-legend-size-category-line--step\\"><div class=\\"as-legend-size-category-line--line-wrapper\\" style=\\"height: 48px;\\"><div class=\\"as-legend-size-category-line--line\\" style=\\"height: 14px; width: 48px; border-top-style: solid; border-top-width: 14px;\\"></div></div><span class=\\"as-legend-size-category-line--label\\">polygon label</span></div></div></as-legend-size-category-line>"`;

exports[`as-legend-size-category-line Rendering renders line styles 1`] = `"<as-legend-size-category-line class=\\"hydrated\\"> <div class=\\"as-legend-size-category-line--steps as-legend-size-category-line--vertical\\"><div class=\\"as-legend-size-category-line--step\\"><div class=\\"as-legend-size-category-line--line-wrapper\\" style=\\"width: 48px;\\"><div class=\\"as-legend-size-category-line--line\\" style=\\"height: 10px; width: 48px; border-top-style: dashed; border-top-width: 10px;\\"></div></div><span class=\\"as-legend-size-category-line--label\\">point label</span></div><div class=\\"as-legend-size-category-line--step\\"><div class=\\"as-legend-size-category-line--line-wrapper\\" style=\\"width: 48px;\\"><div class=\\"as-legend-size-category-line--line\\" style=\\"height: 12px; width: 48px; border-top-style: dotted; border-top-width: 12px;\\"></div></div><span class=\\"as-legend-size-category-line--label\\">line label</span></div><div class=\\"as-legend-size-category-line--step\\"><div class=\\"as-legend-size-category-line--line-wrapper\\" style=\\"width: 48px;\\"><div class=\\"as-legend-size-category-line--line\\" style=\\"height: 14px; width: 48px; border-top-style: dashed; border-top-width: 14px;\\"></div></div><span class=\\"as-legend-size-category-line--label\\">polygon label</span></div></div></as-legend-size-category-line>"`;
